fuel/datasets/hdf5.py

```diff
@@ -411,21 +616,27 @@ def _in_memory_get_data(self, state=None, request=None):
         return data, shapes
 
     def _out_of_memory_get_data(self, state=None, request=None):
+        if not isinstance(request, (slice, list)):
```

Comment:
Do we still have iteration schemes that produce slices? They are not save, because range checks doesn't work for them. If we don't have schemes, I would consider drop the support for slices here.